pkg/config/validation/validation.go

```diff
@@ -3060,12 +3060,8 @@ var ValidateServiceEntry = RegisterValidateFunc("ValidateServiceEntry",
 					"field values", serviceEntry.Resolution))
 			}
 			for _, port := range serviceEntry.Ports {
-				if port == nil {
-					errs = AppendValidation(errs, errors.New("service entry port may not be null"))
-					continue
-				}
-				if protocol.Parse(port.Protocol) != protocol.HTTP {
-					errs = AppendValidation(errs, fmt.Errorf("only HTTP protocol is supported for resolution type %s", serviceEntry.Resolution))
+				if protocol.Parse(port.Protocol) != protocol.HTTP && protocol.Parse(port.Protocol) != protocol.TLS {
```

Comment:
No need to parse the protocol twice; save it to a variable

Reply:
Missed this, thanks